Sources/Widgets/Widgets3D/ResliceCursorWidget/behavior.js

```diff
@@ -68,6 +72,110 @@ export default function widgetBehavior(publicAPI, model) {
+    const imageBounds = image.getBounds();
+    const bounds = vtkBoundingBox.newInstance({ bounds: imageBounds });
+
+    // Get the normal use to translate the center along
+    const dirProj = calldata.pokedRenderer
+      .getRenderWindow()
+      .getRenderers()[0]
+      .getActiveCamera()
+      .getDirectionOfProjection();
+
+    // Direction of the projection is the inverse of what we want
+    const normal = vtkMath.multiplyScalar(dirProj, -1);
+
+    const otherViews = getOtherViews(normal, model.widgetState);
+
+    // Define the potentially new center
+    let newCenter = [
+      oldCenter[0] + direction * normal[0] * imageSpacing[0],
+      oldCenter[1] + direction * normal[1] * imageSpacing[1],
+      oldCenter[2] + direction * normal[2] * imageSpacing[2],
+    ];
+    const newCenterInsideBounds = bounds.containsPoint(...newCenter);
+
+    // The center is out of the image
+    if (!newCenterInsideBounds) {
+      // Find which plane is out of the bounds
+      const isCurrentPlaneOutOfBounds =
+        !vtkBox.intersectPlane(imageBounds, newCenter, normal) &&
+        vtkBox.intersectPlane(imageBounds, oldCenter, normal);
+
+      const otherViewsPlaneOutOfBounds = otherViews.map(
+        (view) =>
+          !vtkBox.intersectPlane(imageBounds, newCenter, view.normal) &&
+          vtkBox.intersectPlane(imageBounds, oldCenter, view.normal)
+      );
+      const otherPlanesAreOutsideBounds = otherViewsPlaneOutOfBounds.every(
+        (value) => value
+      );
+
+      // Basic clamping of the newCenter to the image bounds
+      if (!otherPlanesAreOutsideBounds && isCurrentPlaneOutOfBounds) {
+        // Find new reslice cursor center in local coordinates and round it to be on a valid slice
+        const imageOrigin = image.getOrigin();
+        const localRoundedCenter = oldCenter.map(
+          (value, index) =>
+            Math.round(value - imageOrigin[index]) / imageSpacing[index]
+        );
+
+        // Get it back in world coordinates
+        newCenter = localRoundedCenter.map(
+          (value, index) => value * imageSpacing[index] + imageOrigin[index]
+        );
+      } else {
+        // Need to find which axis will be out of the bounds
+        for (let i = 0; i < otherViews.length; i++) {
+          const isOut = otherViewsPlaneOutOfBounds[i];
+          if (isOut) {
+            const outViewName = otherViews[i].name;
+            const displayViewOutOfBounds =
+              i === 0 ? otherViews[1].name : otherViews[0].name;
+            // Get the correct line that will be used to find the new center
+            // by computing the intersection between current plane and the line
+            const axisName = `Axis${outViewName}in${displayViewOutOfBounds}`;
+            const axis = model.widgetState[`get${axisName}`]();
+            newCenter = vtkPlane.intersectWithLine(
+              axis.getPoint1(),
+              axis.getPoint2(),
+              newCenter,
+              normal
+            ).x;
```

Comment:
isn't `axis` within the plane defined by oldCenter and normal ? 
If so, how can you get an intersection between  the line `axis` and oldCenter and normal ?

Reply:
No. When you enter into the else, it means that only one of the two other planes (not the current one) is out of the bounds.
The cas when it's the current view which is out of the bounds is manage in the if.